Sources/WeaverCodeGen/SwiftGenerator.swift

```diff
@@ -20,11 +24,15 @@ public final class SwiftGenerator {
 
     private let version: String
     
-    public init(dependencyGraph: DependencyGraph,
+    public init(mainOutputPath: Path,
```

Comment:
The generator only uses the name (`lastComponent`). Let's pass that instead of the entire paths to remove confusion.